### PR TITLE
add end to end test

### PR DESCRIPTION
**Pull request by olivesdad**
add end to end testing for 
"as an instructor, I can add an assignment for a course I'm teaching"

### CODE REVIEW

File: cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            //delete the entry
+            assignmentRepository.delete(a);
+            //make sure its gone
+            assertNull(assignmentRepository.findByNameAndCourseId(TEST_ASSIGNMENT_NAME, c).orElse(null));
+
+
+        } catch (Exception ex) {
+            throw ex;
+        } finally {
+            driver.quit();
```
Comment (itsOwlbit) [inline cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java]: Are we supposed to have driver.close()?

File: cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    AssignmentRepository assignmentRepository;
+
+    @Test
+    public void createAssignment() throws Exception {
+        //$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+        // --> (gradebook service) as an Instructor I can create a new assignment for a course that I teach <--
+        //$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+
+        // START THE THINGS///
+        System.setProperty("webdriver.firefox.driver", CHROME_DRIVER_FILE_LOCATION);
```
Comment (itsOwlbit) [inline cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java:76]: oh, you are using firefox.  :)

**Comment by itsOwlbit** [comment]
Other than the possible driver.close() before driver.quit()...

```
 --------
<  LGTM  >
 --------
      \    “___” 
       \  { O,O }   
          (    (\
         __"___"__
         _________
```

P.S. - See the "Errors and fixes for End to End test exercise" announcement and point #3.  Mine worked fine without the driver.close(), but I added on mine anyway just in case.  Your call.

**Review by itsOwlbit — APPROVED** [review]
```
 -------------------------------------
< I approve this message... and code >
 -------------------------------------
      \    “___” 
       \  { O,O }   
          (    (\
         __"___"__
         _________
```

**Comment by olivesdad** [comment]
```
                                                         
                                                         
            i        .Gt                             ;f. 
           LE       j#W: GEEEEEEEL        ..       : i##:
          L#E     ;K#f   ,;;L#K;;.       ,W,     .Et i##:
         G#W.   .G#D.       t#E         t##,    ,W#t i##:
        D#K.   j#K;         t#E        L###,   j###t i##:
       E#K.  ,K#f   ,GD;    t#E      .E#j##,  G#fE#t i##:
     .E#E.    j#Wi   E#t    t#E     ;WW; ##,:K#i E#t i##:
    .K#E       .G#D: E#t    t#E    j#E.  ##f#W,  E#t i##:
   .K#D          ,K#fK#t    t#E  .D#L    ###K:   E#t i#W.
  .W#G             j###t    t#E :K#t     ##D.    E#t ,i. 
 :W##########Wt     .G#t     fE ...      #G      ..  :G#:
 :,,,,,,,,,,,,,.      ;;      :          j           iKt 
                                                         
```

File: cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java (hunk 1 above)
Comment (olivesdad) [reply cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java]: Thank you I put it

File: cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java (hunk 2 above)
Comment (olivesdad) [reply cst438_gradebook/src/test/java/com/cst438/EndToEndGradeBookTest.java:76]: 🦊

**Pull request merged**